# Analyzing Amazon Products

## Duck Products

In this project, we will look at an Amazon products dataset in order to get a general as to what proportion of products are duck products on the Amazon platform.

We will start by analyzing the dataset headings to see what it contains.

In [1]:
from csv import reader

#Starting with the Google Play data set
opened_file = open("amazon_products_sample.csv", encoding='utf-8')
read_file = reader(opened_file)
amazon_prods = list(read_file)
prods_heading = amazon_prods[0]
amazon_prods = amazon_prods[1:]
index = -1
for elem in prods_heading:
    index += 1
    print(index, elem)

0 Uniq Id
1 Product Name
2 Brand Name
3 Asin
4 Category
5 Upc Ean Code
6 List Price
7 Selling Price
8 Quantity
9 Model Number
10 About Product
11 Product Specification
12 Technical Details
13 Shipping Weight
14 Product Dimensions
15 Image
16 Variants
17 Sku
18 Product Url
19 Stock
20 Product Details
21 Dimensions
22 Color
23 Ingredients
24 Direction To Use
25 Is Amazon Seller
26 Size Quantity Variant
27 Product Description


For the purposes of this simple project, we can remove several of the categories above. However, we will just remove any empty strings in the product description along with the images column since it adds a lot of cluster.

In [2]:
for prod in amazon_prods:
    for detail in prod:
        if "https://images-na.ssl" in detail or detail == "" or detail == '' or detail == None:
            prod.remove(detail)
            
print(amazon_prods[0])

['4c69b61db1fc16e7013b43fc926e502d', 'DB Longboards CoreFlex Crossbow 41" Bamboo Fiberglass Longboard Complete', 'Sports & Outdoors | Outdoor Recreation | Skates, Skateboards & Scooters | Skateboarding | Standard Skateboards & Longboards | Longboards', '$237.68', "Make sure this fits by entering your model number. | RESPONSIVE FLEX: The Crossbow features a bamboo core encased in triaxial fiberglass and HD plastic for a responsive flex pattern that’s second to none. Pumping & carving have never been so satisfying! Flex 2 is recommended for people 120 to 170 pounds. | COREFLEX TECH: CoreFlex construction is water resistant, impact resistant, scratch resistant and has a flex like you won’t believe. These boards combine fiberglass, epoxy, HD plastic and bamboo to create a perfect blend of performance and strength. | INSPIRED BY THE NORTHWEST: Our founding ideal is chasing adventure & riding the best boards possible, inspired by the hills, waves, beaches & mountains all around our headquart

The name of the products is always the second element, but the other indices of the list are not not even since some products have specifications that are not available to others. Therefore, we will focus on the name of the product as our main filter.

We will create 3 lists to store all the duck products:
1. Toy Duck Products: It seems that most products involving ducks are toys. Therefore, we will put those as a separate list.
2. Fashion Duck Products: This will encompass all the duck products that fall within the Clothing, Costumes, or accessories categories.
3. Relevant Duck Products: Since some duck products are not relevant to our search, we will only add the relevant ones to this list so we can maximize the accuracy of this search.

In [3]:
duck_prods = []
duck_prods_name = []
toy_duck_prods = []
tdp_name = [] # Names of Toy Duck Prods
fashion_duck_prods = []
fdp_name = [] # Names of Fashion Duck Prods
rel_duck_prods = [] # Relevant Duck Prods
rdp_name = [] # Names of Relevant Duck Prods

for prod in amazon_prods:
    name = prod[1]
    if "duck" in name or "Duck" in name or "DUCK" in name:
        duck_prods.append(prod)
        duck_prods_name.append(prod[1])
        
for prod in duck_prods:
    if ("clothing" in prod[2] or "Clothing" in prod[2] or "CLOTHING" in prod[2] or
        "costumes" in prod[2] or "Costumes" in prod[2] or "COSTUMES" in prod[2] or
        "accessories" in prod[2] or "Accessories" in prod[2] or "ACCESSORIES" in prod[2]):
        fashion_duck_prods.append(prod)
        fdp_name.append(prod[1])
        rel_duck_prods.append(prod)
        rdp_name.append(prod[1])
        
for prod in duck_prods:
    if ("toys" in prod[2] or "Toys" in prod[2] or "TOYS" in prod[2]):
        toy_duck_prods.append(prod)
        tdp_name.append(prod[1])
        rel_duck_prods.append(prod)
        rdp_name.append(prod[1])
        
print("The number of duck products found are:", len(duck_prods))

The number of duck products found are: 32


We will now manually check the lists created above for any errors, and then we can begin our analysis of the data.

In [4]:
n=0
for name in rdp_name:
    n += 1
    print(n, name)
print("\nThe number of relevant duck products found are:", len(rdp_name))

1 Duck-A-Roo! Kids Memory Game in A Duck-Shaped Box
2 Duck Tales Gizmoduck Kids Costume
3 Safety 1st Rubber TempGuard, Ducky
4 Wind-Up Walking Families - Duck Family
5 Squish-Dee-Lish 5 Pack Slow-Rise Squishies, Series 3 - Pink Sea Shell, Blue Whale, Owl, Piñata, Blue Duck Bendable-Toy-Figures
6 Toy Story Disney/Pixar Interactive True Talkers Bunny and Ducky 2-Pack
7 Swimline Giant Ducky Inflatable Ride-On
8 Rhode Island Novelty 2 Inch Multi-Color Pattern Rubber Duck (12 Piece)
9 Pillow Pets Bunny 16” & Ducky Mini Plush - Disney Toy Story 4 Stuffed Animal Set
10 Melissa & Doug Disney Baby Mickey Mouse and Donald Duck Wooden Stacker Toy (12 pcs)
11 Bella Tunno No Scrubs Wonder Duck
12 GAME Large Derby Duck, Inflatable Pool Float, Balloon Animal Pool Float, Durable Vinyl, Quick-Fill Valves, 175-Pound Capacity, 2 Handles, 48” L x 44” W x 27” T
13 The World of Eric Carle, Duckie Bath Book and Squirty Toy Set
14 2" HALLOWEEN DUCKY MIX
15 Duck-A-Roo! Kids Memory Game in A Duck-Shaped Box
16 

It seems that all the relevant duck products do fit into the criteria we specified. Let's explore the lists seperately.

In [5]:
n=0
for name in tdp_name:
    n += 1
    print(n, name)
print("\nThe number of toy duck products found are:", len(tdp_name))

1 Safety 1st Rubber TempGuard, Ducky
2 Wind-Up Walking Families - Duck Family
3 Squish-Dee-Lish 5 Pack Slow-Rise Squishies, Series 3 - Pink Sea Shell, Blue Whale, Owl, Piñata, Blue Duck Bendable-Toy-Figures
4 Toy Story Disney/Pixar Interactive True Talkers Bunny and Ducky 2-Pack
5 Swimline Giant Ducky Inflatable Ride-On
6 Rhode Island Novelty 2 Inch Multi-Color Pattern Rubber Duck (12 Piece)
7 Pillow Pets Bunny 16” & Ducky Mini Plush - Disney Toy Story 4 Stuffed Animal Set
8 Melissa & Doug Disney Baby Mickey Mouse and Donald Duck Wooden Stacker Toy (12 pcs)
9 Bella Tunno No Scrubs Wonder Duck
10 GAME Large Derby Duck, Inflatable Pool Float, Balloon Animal Pool Float, Durable Vinyl, Quick-Fill Valves, 175-Pound Capacity, 2 Handles, 48” L x 44” W x 27” T
11 The World of Eric Carle, Duckie Bath Book and Squirty Toy Set
12 2" HALLOWEEN DUCKY MIX
13 Duck-A-Roo! Kids Memory Game in A Duck-Shaped Box
14 Mega Construx Pokémon Psyduck
15 Simba ABC Bathing Ducks (4 Piece) Bathing Ducks Bath Toy


In [6]:
n=0
for name in fdp_name:
    n += 1
    print(n, name)
print("\nThe number of fashion duck products found are:", len(fdp_name))

1 Duck-A-Roo! Kids Memory Game in A Duck-Shaped Box
2 Duck Tales Gizmoduck Kids Costume

The number of fashion duck products found are: 2


Given that all the data we found is accurate, we can now start our analysis.

In [7]:
print("Total number of products in sample:", len(amazon_prods))
print("Total number of duck products in sample:", len(rdp_name))

Total number of products in sample: 10002
Total number of duck products in sample: 30


In [8]:
prop = len(rdp_name)/len(amazon_prods)
print("The proportion of duck products is:", prop)
print("The percentage of duck products is:", prop*100)

The proportion of duck products is: 0.002999400119976005
The percentage of duck products is: 0.2999400119976005


In [9]:
prop_toy = len(tdp_name)/len(amazon_prods)
print("The proportion of duck products is:", prop_toy)
print("The percentage of duck products is:", prop_toy*100)

The proportion of duck products is: 0.0027994401119776045
The percentage of duck products is: 0.27994401119776047


In [10]:
prop_fashion = len(fdp_name)/len(amazon_prods)
print("The proportion of duck products is:", prop_fashion)
print("The percentage of duck products is:", prop_fashion*100)

The proportion of duck products is: 0.0001999600079984003
The percentage of duck products is: 0.01999600079984003


We can see that the percentage of duck products overall that fall within the categories we previously specified is approximately 0.30%. \~0.28% of those are toys and \~0.02% are fashion products.

## Conclusions

In this project we attempted to find the proportions and percentages of duck products on the Amazon website using a sample dataset to draw our conclusions.

We concluded that there are approximate 0.30% duck products on the Amazon website, 0.28% of those being toys and 0.02% being fashion products.

Since that percentage seems small, let's try putting things into perspective. Overall, Amazon has approximately around 12 million items. Taking 0.30% of that, we get that there are approximately 36,000 duck products on Amazon. 33,600 of those products are likely to be toys, and 2,400 products are fashion products (accessories, clothes, or costumes).

Finally, an important disclaimer:
The [dataset](https://data.world/promptcloud/amazon-product-dataset-2020/workspace/file?filename=marketing_sample_for_amazon_com-ecommerce__20200101_20200131__10k_data.csv%2Fhome%2Fsdf%2Fmarketing_sample_for_amazon_com-ecommerce__20200101_20200131__10k_data.csv) (open link in new tab to see dataset) used in this project has been deprecated, and we are unsure how accurate the information from it is. However, we can safely assume that these rough approximations hold to this day, even if the data is outdated. It is important to reseasrch further in order to obtain more accurate data.

| Duck Products | Percentage of All Products | Approximate Quantity on Website |
| :--- | --- | --- |
| Toys | 0.28% | 33,600 |
| Fashion | 0.02% | 2,400 |
| All | 0.30% | 36,000 |